external-crates/move/crates/move-analyzer/src/analysis/typing_analysis.rs

```diff
@@ -461,6 +562,77 @@ impl TypingAnalysisContext<'_> {
             self.visit_exp(args);
         }
     }
+
+    fn process_match_patterm(&mut self, match_pat: &mut T::MatchPattern) {
+        use T::UnannotatedPat_ as UA;
+
+        self.visit_type(None, &mut match_pat.ty);
+        match &mut match_pat.pat.value {
+            UA::Variant(mident, name, vname, tyargs, fields)
+            | UA::BorrowVariant(_, mident, name, vname, tyargs, fields) => {
+                self.add_datatype_use_def(mident, name);
+                self.add_variant_use_def(mident, name, vname);
+                tyargs.iter_mut().for_each(|t| self.visit_type(None, t));
+                for (fpos, fname, (_, (_, pat))) in fields.iter_mut() {
+                    if self.compiler_info.ellipsis_binders.get(&fpos).is_none() {
```

Comment:
@cgswords, I only use the location of identifiers that need to be "hidden" as they are part of (internally expanded ellipsis). From what I understand these identifiers get the same location as the ellipsis so it should work (and indeed does in practice, at least on examples I tried) but I wonder if you can see something that could break it

Reply:
We don't really have location for ellipsis but I guess I could see if ANY identifier amongst binders is annotated with `EllipsisMatchEntries`. One problem is that I don't think we can tell at naming if a given variant has positional or named fields so it's unclear which variant of the `EllipsisMatchEntries` enum to check against.. Hence the idea of checking location of each identifier for the IDE annotation

Reply:
I guess my thinking is that we would want to associate the pattern with this entry so that we could offer users to "expand" the ellipses. That won't help us much in the positional case, however, and we'll still need to do something like this.

Reply:
> I guess my thinking is that we would want to associate the pattern with this entry so that we could offer users to "expand" the ellipses.

This may not be a bad idea to show ellipsis expansion when hovering over the wild card (not in this PR, though, I think). I will keep the current "hidden" identifiers handling as is then unless I misunderstood and you do like me to change it